libs/pika/executors/include/pika/executors/limiting_scheduler.hpp

```diff
@@ -0,0 +1,188 @@
+//  Copyright (c) 2023 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <pika/config.hpp>
+#include <pika/assert.hpp>
+#include <pika/coroutines/thread_enums.hpp>
+#include <pika/debugging/print.hpp>
+#include <pika/errors/try_catch_exception_ptr.hpp>
+#include <pika/execution/algorithms/execute.hpp>
+#include <pika/execution/algorithms/schedule_from.hpp>
+#include <pika/execution/executors/execution_parameters.hpp>
+#include <pika/execution_base/receiver.hpp>
+#include <pika/execution_base/sender.hpp>
+#include <pika/synchronization/counting_semaphore.hpp>
+#include <pika/threading_base/annotated_function.hpp>
+#include <pika/threading_base/register_thread.hpp>
+#include <pika/threading_base/thread_description.hpp>
+
+#include <cstddef>
+#include <exception>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+namespace pika::execution::experimental {
+
+    // -----------------------------------------------------------------
+    // by convention the title is 7 chars (for alignment)
+    // a debug level of N shows messages with level 1..N
+    using namespace pika::debug::detail;
+    template <int Level>
+    static print_threshold<Level, 0> lsc_debug("SCLIMIT");
+
+    template <typename OnScheduler>
+    struct limiting_scheduler
```

Comment:
Would this be ok as well:
```suggestion
    template <typename Scheduler>
    struct limiting_scheduler
```
?

Reply:
I can't rename the OnScheduler to Scheduler, cos, that name is used for the Op state and internal sender template params. But I'll call it WrappedScheduler for the time being.